domain/schema/controller.go

```diff
@@ -771,20 +763,72 @@ INSERT INTO permission_object_access VALUES
 -- We will need to remove/replace it in the event of change
 CREATE TABLE permission (
     uuid               TEXT PRIMARY KEY,
-    permission_type_id INT NOT NULL,
+    access_type_id     INT NOT NULL,
+    object_type_id     INT NOT NULL,
     grant_on  		   TEXT NOT NULL, -- name or uuid of the object
     grant_to           TEXT NOT NULL,
     CONSTRAINT         fk_permission_user_uuid
         FOREIGN KEY    (grant_to)
         REFERENCES     user(uuid),
-    CONSTRAINT         fk_permission_access_type
-        FOREIGN KEY    (permission_type_id)
-        REFERENCES     permission_access_type(id)
+    CONSTRAINT      fk_permission_access_type
```

Comment:
These should be one constraint:
```sql
        FOREIGN KEY (access_type_id, object_type_id)
        REFERENCES  permission_object_access(access_type_id, object_type_id)
```
That way we constrain it to allowed combinations.